### ISSUE TITLE
when creating regional filter and trying to match I get an "Error with Datasets, check again"

### ISSUE THREAD

**Issue by sinairusinek** [UHaifa-IS/whgazetteer-mehdie#150]
this happend both with KimaMisrPlusMatched and KimaMisrunmapped
<img width="1137" alt="Screen Shot 2024-01-09 at 10 18 38" src="https://github.com/UHaifa-IS/whgazetteer-mehdie/assets/4381316/bf289e28-01a1-4edf-981b-7fbd820dfb03">

**Comment by tomersagi** [comment]
ok, we will investigate

**Comment by tomersagi** [comment]
@sinairusinek  Please try again

**Comment by sinairusinek** [comment]
I ran it,  did not get the error message, but it looks like the polygon filtering did not work: I got the following result, for example (not in Misr):
<img width="1177" alt="Screen Shot 2024-01-26 at 15 16 16" src="https://github.com/UHaifa-IS/whgazetteer-mehdie/assets/4381316/ef43aa6b-fd21-4aae-a6ba-e43505461c46">

**Comment by tomersagi** [comment]
Can you explain the expected behavior and why the results violate this in a more explicit way?

**Comment by sinairusinek** [comment]
I created a polygon filter that is supposed to limit the matching only to places which have locations within that polygon:
![Screen Shot 2024-01-27 at 10 22 49](https://github.com/UHaifa-IS/whgazetteer-mehdie/assets/4381316/e39cd8f9-0265-4f1a-957c-fabc9f504d8f)
However most of the offered matches are from outside this area.

**Comment by sinairusinek** [comment]
I'll just add that I did it as a trial - this feature could have saved us the splitting of our datasets to regions. Now that we did split them we can progress without it, so it is not urgent,  but it is definitely on our wish list.

**Comment by tomersagi** [comment]
So the procedure is to filter both datasets using this polygon before
matching?

**Comment by sinairusinek** [comment]
No, it would be sufficient to filter the target dataset. The typical use case is that you have a dataset of **unmapped** names from some region, and you have a large mapped dataset, you select the relevant region and only try to match to your dataset  the items from the large dataset that are in that region.

**Comment by tomersagi** [comment]
ok, got it.
